### DISCUSSION TITLE
No connection after running sudo apt-get install network-manager -y

### DISCUSSION THREAD

**Discussion by cottec** [Guilouz/Klipper-Flsun-Speeder-Pad#178] [Q&A] [answered]
So really really weird issue here.
I can follow every single step, i have a stable and fast wifi.

Right after I follow the step to install the network manager and wait like for 1 hour to get the error message, I do the unplug and plug.
From that moment on I wont get an IP on the same network I used before.
It's showing as connected in my router but it doesnt get an IP.
I tried several times now, I'm always getting the issue after the installation of network manager.
Interestingly, if I choose an old LTE router I've got near me, I can connect to this one.

But even with a completed setup it's impossible to return to the standard wifi. changing ssid, channels etc doesnt change anything.
From this stable connection before 
![image](https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/assets/78232726/26ae42f2-8019-4b7f-a3e7-5c827f32924c)
to this afterwards
![image](https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/assets/78232726/9aa4b0cc-f7a4-4c5e-9230-5db48631f727)

**Comment by lin-ycv** [comment]
I also think there's some issues with the network manager, sometimes i find the printer connects to the router, shows an ip, but i'm unable to ssh or access mainsail, i'd have to reconnect using the speeder pad.

**Reply by SckyzO** [reply]
I have the same issue. And i need to restart the printer or just reconnect printier to wifi via klipperscreen. 

Another point, Wifi with 2 bands:  2.4Ghz and 5Ghz, gives me problems. so now I use an ESP32 to be a repeater/extender from my wifi, with only one access point in 2.4Ghz, dedicated to my printers, and with a dedicated SSID (Printer_wifi_2.4 for example)

**Reply by cottec** [reply]
i dont have any clue, i updated from python 2.7 klipper to 3 and since then I cant connect to one specific network, another one is working just fine, how can that be? I didnt change anything on the router

**Reply by lin-ycv** [reply]
fixed my issue by creating a script that automatically restarts network manager on startup
https://youtu.be/SuXGwg9b_1Q

**Comment by fzechert** [comment]
The second I install network manager, I can no longer connect to any network. I tried a couple different networks. My usual one, guest Wifi, a Hotspot etc. With no luck.

Any suggestions here on how to continue the install, if it is impossible to connect to any network after `sudo apt-get install network-manager -y`

**Reply by cottec** [reply, marked as answer]
Probably speeder pad running out of RAM+ some weird incompatibility with our specific hardware.
Just skip the step and you'll be fine. Only downside is the missing WiFi manager. But as long as you don't change WiFi every week you don't need to worry... WiFi can be changed in /etc/wpa_suppplicant/wpa_supplicant.conf by hand

**Reply by fzechert** [reply]
Yeah, it is a little bit difficult to diagnose network errors, if every time you reproduce them, you are locked out of the system because the network connection broke.

I am seriously thinking about soldering an ethernet port to the board (does someone know the part number of the RJ45 jack needed - just by coincidence :zap:). Or connecting to the serial console.

In the meantime I managed to connect to a hotspot opened by quite an old phone I found in one of my drawers.
But it only leads to more problems. As the system is now super slow, and the KlipperScreen constantly throws errors at me:

![3](https://github.com/user-attachments/assets/161d3cce-4483-47b9-a04f-365b9fd9c61e)
![2](https://github.com/user-attachments/assets/a1c8e04f-6b9e-41c5-a732-e4ee8c983828)
![1](https://github.com/user-attachments/assets/d154447e-f795-4470-82f1-289c0d18107c)

I even enabled systemd and networkmanager debug logs. But there is nothing obvious I can find in the logs.
Played around with disabling and enabling  a couple of systemd services, and deleting already existing connection profiles from the `/etc/NetworkManager/` folder and `/etc/wpa_supplicant/wpa_supplicant.conf` files.

I even tried swapping wpa_supplicant for iwd to manage the wifi. But iwd doesn't even start on the SpeederPad. It lists supported kernel encryption methods and then just crashes.

So far no luck :(.

For now I will go with the non-NetworkManager route as you suggested. Thanks.

**Reply by fzechert** [reply]
Just to confirm this. I have now reinstalled from scratch and skipped all the NetworkManager related stuff.
It works fine now. no slowdowns, no lag, no errors on the screen, stable WiFi connection to my normal network.

Only drawback, of course: the network panel is no longer working. But that is a sacrifice i am very willing to make. After fighting with this for 2 days straight I am taking what I am getting.

@cottec Thanks for the suggestion to just skip the NetworkManager.

**Reply by kolergy** [reply]
I had the same issue and spent 4 hours on this re-flashing the machine each time, I would suggest to mark this section of the Wiki as non functional until a fix is found

**Reply by cottec** [reply]
Glad I'm not the only one (which I seemed to be for a long time)

**Reply by SckyzO** [reply]
try to use this script : https://youtu.be/SuXGwg9b_1Q

**Reply by fzechert** [reply]
I tried in one of my many attempts. It did not help.

**Reply by Dark-Noir** [reply]
Same issue here, it is no longer possible to connect to a Wifi network after installing the network manager, @Guilouz please fix the wiki ^^'

**Reply by j-Nimble** [reply]
I used the following command, and ignored all other network related commands on the tutorial.  This had the side effect of allowing USB Network Adapters to work which didn't previously.

`sudo apt install -y network-manager net-tools ifupdown dhcpcd5 usbutils`

**Comment by Manuelrios509** [comment]
The issue is not with the pad, but with your local WIFI router, I was facing the same issue and had to turn off WPA3 security protocols on my router and go to WPA2 and everything connects like it should, I'm guessing the pad doesn't know how to communicate with newer routers

**Reply by cottec** [reply]
no, thats not true. wifi communication is working excellent before installation of network manager

**Reply by Manuelrios509** [reply]
That's how I got mine to work after updating the network manager, just throwing it out there

**Reply by gaizkitapowers** [reply]
Thank you!! You saved my day!
After some hours reflashing the speeder I suspected something since it conected to my old wifi from the basement but didn´t to my brand new mesh. I have arranged the wifi for guests in WPA2 mode without losing WPA3 benefits for the main one.
I couldn´t believe to not being warned as @kolergy suggested and now I can understand why.

**Comment by brdi** [comment]
On my Speederpad the Wifi is handled by a Broadcom-chip BCM43438A1 with a proprietry driver.
This chip has serious limitations. I discoverd that it ignores the wifi region settings. So perhaps it has other limitations as well that should be taken into account:
A quick survey revealed the following issues:

Network related
- ❌ No 5GHz support
- ❌ No 802.11ac (Wifi5) support
- ❌ No support for MIMO speed improvement (Multiple antenna's)
- ⚠️ Limited supprot for WiFi-Direct / P2P (Depending on driver)
- ⚠️ Host AP-mode (access point) limited and not stable
- ⚠️ DFS-channels only passive

Security related
-❌ WPA3 (SAE/OWE) not supported
-⚠️ Enterprise (802.1X/RADIUS) not stable
-❌ PMF (Protected Management Frames) |Not supported
-⚠️ WEP / WPA (TKIP) | ⚠️ Legacysupport but Not safe

Driver-gerelateerde issues (Linux)
- Proprietaire driver (brcmfmac, dhd):
    -  Quite imited (feature wise)
    -  Reported problems with regulatory domains, scanning, or stable reconnect
    -  Not fully compatiblel with cfg80211 / nl80211, (regarding hostapd, iw, or wpa_supplicant )
    -  No support for monitor mode or packet injection (only with big patches of alternative firmware (often not available).

**Comment by tkocou** [comment]
Here is the real crux of the problem -  the Ubuntu Linux version is 20.04. The NetworkManager for that release does not know how to talk with the current routers. I did not have the stated problem until I upgraded my router to one of the newer models. Even forcing the wireless security to WPA2 did not resolve the issue. After re-installing the OS multiple times and trying various tricks, I could not get past that error.
The long term solution would be to update Ubuntu to the latest LTS version (24.04). Question is " **Would upgrading Ubuntu Linux to the latest LTS version break the various scripts and patches listed in the Wiki?**"

**Reply by Dark-Noir** [reply]
I tried to upgrade the OS to any newer LTS without success, the system no longer boots after it, but maybe you can figure it out.
I am already sold my machine.

**Comment by tkocou** [comment]
I looked into the **Processing triggers for systemd** hang up. Apparently the postinst file in network-manager installation is causing an error. Because the speeder pad has no means of communication besides ssh, the speeder pad is hung with a disconnect from the network. As _fzechert_ has stated, just ignore the network-manager. As long as your wireless connection does not change, there won't be an issue.
